extension-adinfo/src/adinfo.cpp

```diff
@@ -36,15 +51,45 @@ static const luaL_reg Module_methods[] =
 static void LuaInit(lua_State* L)
 {
     DM_LUA_STACK_CHECK(L, 0);
-    ADInfo_InitAdInfoExt();
     int top = lua_gettop(L);
     luaL_register(L, MODULE_NAME, Module_methods);
     lua_pop(L, 1);
 }
 
+static dmExtension::Result UpdateAdInfo(dmExtension::Params* params)
+{
+    if(g_ADInfoData.m_callback != 0 && g_ADInfoData.m_AdId != 0)
+    {
+        lua_State* L = dmScript::GetCallbackLuaContext(g_ADInfoData.m_callback);
+        DM_LUA_STACK_CHECK(L, 0);
+
+        if (dmScript::SetupCallback(g_ADInfoData.m_callback))
```

Comment:
Do we need to check if the callback is still valid?

I wonder if we should add a call to `dmScript::IsCallbackValid()` inside the `dmScript::SetupCallback()`... 🤔

Reply:
Yes, we probably should do that!

You mean adding dmScript::IsCallbackValid() inside the if {} in adinfo.cpp or in the implementation of dmScript::SetupCallback()?

Reply:
I'm thinking we might add it inside `dmScript::SetupCallback()`
You never want to call it without checking the validity of the callback.